presto-main/src/main/java/com/facebook/presto/sql/planner/planPrinter/WindowOperatorStats.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.planPrinter;
+
+import com.facebook.presto.operator.WindowInfo;
+import com.facebook.presto.util.Mergeable;
+
+class WindowOperatorStats
+        implements Mergeable<WindowOperatorStats>
+{
+    private final int activeDrivers;
+    private final int totalDrivers;
+    private final double positionsInIndexesSumSquaredDiffs;
+    private final double sizeOfIndexesSumSquaredDiffs;
+    private final double indexCountPerDriverSumSquaredDiffs;
+    private final double partitionRowsSumSquaredDiffs;
+    private final double rowCountPerDriverSumSquaredDiffs;
+    private final long totalRowCount;
+    private final long totalIndexesCount;
+    private final long totalPartitionsCount;
+
+    public static WindowOperatorStats create(WindowInfo info)
+    {
+        int activeDrivers = 0;
+        int totalDrivers = 0;
+
+        double partitionRowsSumSquaredDiffs = 0.0;
+        double positionsInIndexesSumSquaredDiffs = 0.0;
+        double sizeOfIndexesSumSquaredDiffs = 0.0;
+        double indexCountPerDriverSumSquaredDiffs = 0.0;
+        double rowCountPerDriverSumSquaredDiffs = 0.0;
+        long totalRowCount = 0;
+        long totalIndexesCount = 0;
+        long totalPartitionsCount = 0;
+
+        double averageNumberOfIndexes = info.getWindowInfos().stream()
+                .filter(windowInfo -> windowInfo.getTotalRowsCount() > 0)
+                .mapToLong(windowInfo -> windowInfo.getNumberOfIndexes())
+                .average()
+                .orElse(Double.NaN);
```

Comment:
Having zero indexes is not allowed state, that's why I wanted to express the wrong of stats collection by simply displaying "NaN" to the user, instead of masking the problem behind, simply displaying wrong number (e.g. `Math.pow(numberOfIndexes - averageNumberOfIndexes, 2)` would return `0` which is not so worrisome as `NaN`, at least at first glance (because after analyzing the output, you'd notice that zero is not legal value there too).